packages/protocols/generation-queue.go

```diff
@@ -0,0 +1,127 @@
+package protocols
+
+import (
+	"errors"
+	"time"
+
+	"github.com/GenesisKernel/go-genesis/packages/model"
+
+	"github.com/GenesisKernel/go-genesis/packages/conf/syspar"
+	log "github.com/sirupsen/logrus"
+)
+
+// QueueChecker allow check queue to generate current block
+type QueueChecker interface {
```

Comment:
And again, what queue does it actually check?
All functions of that interface is about block generation time.
Maybe better name it something like BlockTimeChecker